src/marqo/core/structured_vespa_index.py

```diff
@@ -279,7 +285,50 @@ def _to_vespa_tensor_query(cls, marqo_query: MarqoTensorQuery, marqo_index: Marq
 
     @classmethod
     def _to_vespa_lexical_query(cls, marqo_query: MarqoLexicalQuery, marqo_index: MarqoIndex) -> Dict[str, Any]:
-        raise NotImplementedError()
+        if marqo_query.searchable_attributes is not None:
+            for att in marqo_query.searchable_attributes:
+                if att not in marqo_index.lexically_searchable_fields_names:
+                    raise InvalidFieldNameError(
+                        f'Index {marqo_index.name} has no lexically searchable field {att}. '
+                        f'Available lexically searchable fields are '
+                        f'{", ".join(marqo_index.lexically_searchable_fields_names)}'
+                    )
+            fields_to_search = marqo_query.searchable_attributes
+        else:
+            fields_to_search = marqo_index.lexical_fields_names
+
+        lexical_term = cls._get_lexical_search_term(marqo_query, marqo_index)
+        filter_term = cls._get_filter_term(marqo_query)
+        if filter_term:
+            filter_term = f' AND {filter_term}'
+        else:
+            filter_term = ''
+
+        select_attributes = cls._get_select_attributes(marqo_query)
+        summary = cls._SUMMARY_ALL_VECTOR if marqo_query.expose_facets else cls._SUMMARY_ALL_NON_VECTOR
+        score_modifiers = cls._get_score_modifiers(marqo_query, marqo_index)
+        ranking = cls._RANK_PROFILE_BM25_MODIFIERS if score_modifiers \
+            else cls._RANK_PROFILE_BM25
+
+        query_inputs = {}
+        query_inputs.update({
+            f: 1 for f in fields_to_search
+        })
+        if score_modifiers:
+            query_inputs.update(score_modifiers)
+
+        query = {
```

Comment:
Would it be good to have a pedantic class to better manage the Vespa query dict?

Reply:
Yes, I think there are a few things that require pydantic models. We'll keep this as tech debt for now and add later